diff --git a/docroot/.htaccess b/docroot/.htaccess
index 83cb71a..6fad7ae 100644
--- a/docroot/.htaccess
+++ b/docroot/.htaccess
@@ -69,6 +69,49 @@ AddEncoding gzip svgz
 <IfModule mod_rewrite.c>
   RewriteEngine on
 
+  # Block access to specific files/paths to all users except stanford IP's.
+  # Each condition blocks unwanted access to all users at the apache level.
+  # This prevents drupal from bootstrapping. Most of these conditions are
+  # requested by bots or by user's cached requests.
+  RewriteCond %{REQUEST_URI} ^/simplesaml [OR]
+  RewriteCond %{REQUEST_URI} ^/update.php$ [OR]
+  RewriteCond %{REQUEST_URI} ^/core/install.php [OR]
+  RewriteCond %{REQUEST_URI} "/wp-(admin|includes|cron\.php|config\.php|login\.php|signup\.php)" [OR]
+  RewriteCond %{REQUEST_URI} ^/wp-content/.*(.js|.css|.jpg|.jpeg|.png|.pdf)$
+  # Only need to block on Acquia's environment. Local isn't a concern.
+  RewriteCond %{ENV:AH_SITE_ENVIRONMENT} (dev|test|prod|live)$
+  # The following path is always required to log into the site.
+  RewriteCond %{REQUEST_URI} !^/simplesaml/module.php/saml/sp/saml2-acs.php/default-sp
+  RewriteCond %{ENV:AH_Client_IP} !^127.0.0.1$
+  RewriteCond %{ENV:AH_Client_IP} !^171.66.21.182$
+  # Block the above conditions on acquia infrastructure.
+  # See https://httpd.apache.org/docs/2.4/rewrite/flags.html#flag_f.
+  RewriteRule .* - [F]
+
+  # Rewrite URLs of the form foo.cardinalsites.acsitefactory.com to
+  # https://foo.sites.stanford.edu
+  # See https://stanfordits.atlassian.net/browse/SITES-129.
+  RewriteCond "%{HTTP_HOST}"   "^(.*)\.cardinalsites\.acsitefactory\.com" [NC]
+  RewriteRule "^/?(.*)"        "https://%1.sites.stanford.edu/$1" [L,R,NE]
+
+  # Test environment.
+  RewriteCond "%{HTTP_HOST}"   "^(.*)\.test-cardinalsites\.acsitefactory\.com" [NC]
+  RewriteRule "^/?(.*)"        "https://%1-test.sites.stanford.edu/$1" [L,R,NE]
+
+  # Dev environment.
+  RewriteCond "%{HTTP_HOST}"   "^(.*)\.dev-cardinalsites\.acsitefactory\.com" [NC]
+  RewriteRule "^/?(.*)"        "https://%1-dev.sites.stanford.edu/$1" [L,R,NE]
+
+  # Remote Administration environment.
+  RewriteCond "%{HTTP_HOST}"   "^(.*)\.acqra-cardinalsites\.acsitefactory\.com" [NC]
+  RewriteRule "^/?(.*)"        "https://%1-acqra.sites.stanford.edu/$1" [L,R,NE]
+
+  # Redirect HTTP to HTTPS on cloud
+  RewriteCond %{HTTP:X-Forwarded-Proto} !https
+  RewriteCond %{ENV:AH_SITE_ENVIRONMENT} ^.*(dev|test|prod|live)$ [NC,OR]
+  RewriteCond %{ENV:AH_SITE_ENVIRONMENT} ^ode.*$ [NC]
+  RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [NE,L,R=301]
+
   # Set "protossl" to "s" if we were accessed via https://.  This is used later
   # if you enable "www." stripping or enforcement, in order to ensure that
   # you don't bounce between http and https.
@@ -146,10 +189,14 @@ AddEncoding gzip svgz
   RewriteCond %{REQUEST_URI} !/core/[^/]*\.php$
   # Allow access to test-specific PHP files:
   RewriteCond %{REQUEST_URI} !/core/modules/system/tests/https?.php
+  # Allow access to simplesaml paths.
+  RewriteCond %{REQUEST_URI} !^/simplesaml
   # Allow access to Statistics module's custom front controller.
   # Copy and adapt this rule to directly execute PHP files in contributed or
   # custom modules or to run another PHP application in the same directory.
   RewriteCond %{REQUEST_URI} !/core/modules/statistics/statistics.php$
+  # ACSF requirement: allow access to apc_rebuild.php.
+  RewriteCond %{REQUEST_URI} !/sites/g/apc_rebuild.php$
   # Deny access to any other PHP files that do not match the rules above.
   # Specifically, disallow autoload.php from being served directly.
   RewriteRule "^(.+/.*|autoload)\.php($|/)" - [F]
